apps/sched/sched.js

```diff
@@ -1,10 +1,81 @@
+/* eslint-disable no-unused-vars */
+
 // Chances are boot0.js got run already and scheduled *another*
 // 'load(sched.js)' - so let's remove it first!
 if (Bangle.SCHED) {
   clearInterval(Bangle.SCHED);
   delete Bangle.SCHED;
 }
 
+function formatMS(ms) {
+  if (ms < 60000) {
+    // less than a minute → show seconds
+    return Math.round(ms / 1000) + "s";
+  } else {
+    // one minute or more → show minutes
+    return Math.round(ms / 60000) + "m";
+  }
+}
+
+function showSnoozeMenu(alarm){
+  const alarmIndex = alarms.indexOf(alarm);
+  const settings = require("sched").getSettings();
+  let buzzCount = 0;
+  let buttons=
+  Bangle.buzz(40);
+  if(alarm.timer){
+    
+    let timerLength=alarm.timer
+    let buttons={ "15s": 15, "30s":30,"1m":60 ,"2m":120,"5m":360};
+    let formattedLength = formatMS(timerLength)+"*";
+    buttons[formattedLength] = Math.round(timerLength/1000);
+    //different button lengths
+    E.showPrompt("Choose snooze length", {
+      title: "Snooze Options",
+      buttons: buttons
+    }).then(function (snoozeTime) {
+      buzzCount = 0;
+
+
+      if (alarm.ot === undefined) {
+        alarm.ot = alarm.t;
+      }
+      let time = new Date();
+      let currentTime = (time.getHours()*3600000)+(time.getMinutes()*60000)+(time.getSeconds()*1000);
+      alarm.t = currentTime + snoozeTime*1000;
+      alarm.t %= 86400000;
+      Bangle.emit("alarmSnooze", alarm);
+
+      // The updated alarm is still a member of 'alarms'
+      // so writing to array writes changes back directly
+      require("sched").setAlarms(alarms);
+      load();
+    });
+  }else{
+    E.showPrompt("Choose snooze length", {
+      title: "Snooze Options",
+      buttons: { "1m": 1, "2m":2,"5m": 5,"10m":10 } 
+    }).then(function (snoozeTime) {
```

Comment:
These two functions are almost the same - perhaps we merge them? Or `.bind()` the difference? I'm happy to add a commit to do it if you like

The difference being:

```diff
      let time = new Date();
      let currentTime = (time.getHours()*3600000)+(time.getMinutes()*60000)+(time.getSeconds()*1000);
-     alarm.t = currentTime + snoozeTime*1000;
+     alarm.t = currentTime + snoozeTime*60000;
      alarm.t %= 86400000;
      Bangle.emit("alarmSnooze", alarm);
```

Reply:
Ah sorry - I didn't explain what I meant properly, we can keep the snooze specifics and menus separate for easy modification still. What do you think of [something like this change](https://github.com/espruino/BangleApps/pull/3978/commits/8a2601bc1dbb0013c96cf5c378afb9a2833f9548)?

Reply:
You're welcome, hope you don't mind me taking that liberty :)